hivemind/server/expert_backend.py

```diff
@@ -1,14 +1,14 @@
-from typing import Dict, Sequence, Any, Tuple, Union
+from typing import Dict, Sequence, Any, Tuple, Union, Callable
 
 import torch
 from torch import nn
 
 from hivemind.server.task_pool import TaskPool
-from hivemind.utils import nested_flatten, nested_pack, nested_compare, nested_map, \
-    BatchTensorDescriptor, DUMMY_BATCH_SIZE
+from hivemind.utils import nested_flatten, nested_pack, nested_compare, nested_map, BatchTensorDescriptor, \
+    DUMMY_BATCH_SIZE
 
 
-class ExpertBackend(nn.Module):
+class ExpertBackend:
```

Comment:
please explain the motivation behind removing nn.module dependence

[the reason it was there: for convenience in backend.cuda() / backend.parameters()]

Reply:
It was never a proper nn.Module to begin with (no `forward`, for instance), and after adding the schedule I concluded that it might be misleading to still treat it as a module, since `backend.state_dict()` will not contain the entire state to be saved

Reply:
[agreed]